Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -195,13 +191,14 @@ void UPSHUDComponent::UpdateProgressionWidgetForPlayer()
 //Is called when local player character is ready to guarantee that they player controller is initialized for the Widget SubSystem
 void UPSHUDComponent::OnLocalCharacterReady_Implementation(APlayerCharacter* Character, int32 CharacterID)
 {
-	// Create widgets now as fast as possible, later we will register them in Widgets Subsystem
-	UWidgetsSubsystem& WidgetsSubsystem = UWidgetsSubsystem::Get(Character);
-	ProgressionMenuWidgetInternal = WidgetsSubsystem.CreateManageableWidgetChecked<UPSMenuWidget>(UPSDataAsset::Get().GetProgressionMenuWidget());
-	ProgressionMenuOverlayWidgetInternal = WidgetsSubsystem.CreateManageableWidgetChecked<UPSOverlayWidget>(UPSDataAsset::Get().GetProgressionOverlayWidget());
+	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	if (PlayerCharacter && !PlayerCharacter->IsLocallyControlled())
```

Comment:
- Dont forget use function arguments, they often contain newest data than trying to obtain it by yourself: use `Character` argument instead of `UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter()`.
- Do early return if chacter is null OR is not local, instead of AND

Before:
```
	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
	if (PlayerCharacter && !PlayerCharacter->IsLocallyControlled())
```

After:
```
	if (!Character || !PlayerCharacter->IsLocallyControlled())
```

Reply:
done with https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/ab4dd01309ec517777311907d9caeaa10759b34e

Reply:
@h4rdmol Next is not finished yet
> Do early return if chacter is null OR is not local, instead of AND

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/9dc2e10b7d1d11f8faa2c1e91a848c39f269dfee

Reply:
It's still wrong and not the same as in my original comment, moreover it will break progression

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/c3ff41c54ac7f90efcee0f655d9acd8156c781a4

Reply:
@h4rdmol this commit introduced unnecessary wrong include:
![image](https://github.com/user-attachments/assets/e4073eca-0c3c-4b21-a19d-dde8dfd23335)

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/c3ff41c54ac7f90efcee0f655d9acd8156c781a4